src/Azure.DataApiBuilder.Mcp/BuiltInTools/ExecuteEntityTool.cs

```diff
@@ -0,0 +1,453 @@
+                }
+
+                // 6) Authorization - Never bypass permissions
+                IAuthorizationResolver authResolver = serviceProvider.GetRequiredService<IAuthorizationResolver>();
+                IHttpContextAccessor httpContextAccessor = serviceProvider.GetRequiredService<IHttpContextAccessor>();
+                HttpContext? httpContext = httpContextAccessor.HttpContext;
+
+                if (!McpAuthorizationHelper.ValidateRoleContext(httpContext, authResolver, out string roleError))
+                {
+                    return McpResponseBuilder.BuildErrorResult("PermissionDenied", roleError, logger);
+                }
+
+                if (!McpAuthorizationHelper.TryResolveAuthorizedRole(
+                    httpContext!,
+                    authResolver,
+                    entity,
+                    EntityActionOperation.Execute,
+                    out string? effectiveRole,
+                    out string authError))
+                {
+                    return McpResponseBuilder.BuildErrorResult("PermissionDenied", authError, logger);
+                }
+
+                // 7) Validate parameters against metadata
+                if (parameters != null && entityConfig.Source.Parameters != null)
+                {
+                    // Validate all provided parameters exist in metadata
+                    foreach (KeyValuePair<string, object?> param in parameters)
+                    {
+                        if (!entityConfig.Source.Parameters.ContainsKey(param.Key))
+                        {
+                            return McpResponseBuilder.BuildErrorResult("InvalidArguments", $"Invalid parameter: {param.Key}", logger);
+                        }
+                    }
+                }
+
+                // 8) Build request payload
+                JsonElement? requestPayloadRoot = null;
+
+                if (parameters?.Count > 0)
+                {
+                    string jsonPayload = JsonSerializer.Serialize(parameters);
+                    using JsonDocument doc = JsonDocument.Parse(jsonPayload);
+                    requestPayloadRoot = doc.RootElement.Clone();
+                }
+
+                // 9) Build stored procedure execution context
+                StoredProcedureRequestContext context = new(
+                    entityName: entity,
+                    dbo: dbObject,
+                    requestPayloadRoot: requestPayloadRoot,
+                    operationType: EntityActionOperation.Execute);
+
+                // First, add user-provided parameters to the context
+                if (requestPayloadRoot != null)
+                {
+                    foreach (JsonProperty property in requestPayloadRoot.Value.EnumerateObject())
+                    {
+                        context.FieldValuePairsInBody[property.Name] = GetParameterValue(property.Value);
```

Comment:
We are deserializing the requestPayloadRoot here into a Dictionary of string, value names. But couldnt we simply have assigned the parsed `parameters` to `FieldValuePaisInBody`? 

I think we are doing unnecessary serialization of `parameters` passed to the MCP endpoint and deserializing them again to conform to the REST request requirements.

Reply:
While this may be fine for now, to improve performance - we should eliminate unnecessary serialization and deserialization.